packages/core/src/shared-priority-cache/shared-cache.ts

```diff
@@ -0,0 +1,143 @@
+import * as lo from 'lodash' // vite/astro went bonkers - not sure why this helps, core has always had lodash in it...
+import { PriorityCache, type Store, type Resource } from "./priority-cache";
+const { uniqueId } = lo
+// goal: we want clients of the cache to experience a type-safe interface -
+// they expect that the things coming out of the cache are the type they expect (what they put in it)
+// this is not strictly true, as the cache is shared, and other clients may use different types
+// also also - there will not be a 1:1 relation between items and
+type CacheInterface<Key, Value extends Record<string, Resource>> = {
```

Comment:
todo: rename generic parameter Key here - like our current cache, we let the users of the cache deal in objects which can require multiple, distinct entries in the cache itself.

Reply:
renamed